### PR TITLE
Add Authorization post

### PR DESCRIPTION
**Pull request by g7ed6e**

### CODE REVIEW

File: _posts/2022-12-07-aspnetcore-authorization-support.markdown [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+# Layout
+layout: post
+title:  "Introducing ASP.NET Core Authorization support and modernization of legacy WCF Authentication and Authorization APis"
+date:   2022-12-07 13:00:00 +0100
+categories: release
+# Author
+author: Guillaume Delahaye (https://github.com/g7ed6e)
+---
+### Introduction
+Next release of CoreWCF will bring support of ASP.NET Core Authorization to allow developers to use ASP.NET Core builtin authentication middleware such as the `Microsoft.AspNetCore.Authentication.JwtBearer` and apply appropriate authorization policies.
```
Comment (mconnew) [inline _posts/2022-12-07-aspnetcore-authorization-support.markdown]: The latest release of....

File: _posts/2022-12-07-aspnetcore-authorization-support.markdown [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+### Builtin attributes support
+When working with ASP.NET Core MVC usually developers use `[Authorize]` and `[AllowAnonymous]` to decorate actions that require specific authorizations.
+#### Authorize support
+To enable a seamless developer experience we brought the ability to decorate `OperationContract` implementation with the ASP.NET Core Authorize attribute. However we introduced the below limitations to suggest developers to embrace the flexible [Policy-based](https://learn.microsoft.com/en-us/aspnet/core/security/authorization/policies?view=aspnetcore-6.0) model based on `IAuthorizationRequirement`.
```
Comment (mconnew) [inline _posts/2022-12-07-aspnetcore-authorization-support.markdown]: Either:  
...to decorate `OperationContract` implementation methods with...  
or  
...to decorate an `OperationContract` implemention with...

File: _posts/2022-12-07-aspnetcore-authorization-support.markdown [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+- `AuthenticationSchemes` property is not supported and will trigger a build warning `COREWCF_0201`.
+- `Roles` property is not supported and will trigger a build warning `COREWCF_0202`.
+
+#### AllowAnonymous support
+We did not bring support of the `[AllowAnonymous]` attribute as we believe that a strong interface segregation between anonymous and secured operations should be set. Moreover supporting this attribute would imply delaying the authentication step in the pipeline leading to potential DDoS vulnerabilities. Decorating an `OperationContract` implementation with `[AllowAnonymous]` will have no effect and will trigger a build warning `COREWCF_0200`.
```
Comment (mconnew) [inline _posts/2022-12-07-aspnetcore-authorization-support.markdown]: DDoS -> DoS  
DDoS generally refers to an attack which relies on flooding the bandwidth of the service (which is why you need lots of clients), or circumventing protections preventing single ip addresses from making too many requests. The nature of the potential vulnerability here doesn't need a distributed attack.

File: _posts/2022-12-07-aspnetcore-authorization-support.markdown [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+### Configuration
+To setup this feature in your CoreWCF application you should follow the below steps. I will assume there that we want to enforce clients are authenticating using a JWT Bearer token issued by an authorization server `https://authorization-server-uri`, the service should be protected by the audience `my-audience` and two policies should be defined, one requiring a scope `read` and another one requiring a scope `write`.
```
Comment (mconnew) [inline _posts/2022-12-07-aspnetcore-authorization-support.markdown]: I'm assuming that we want to enforce clients authenticating using....

File: _posts/2022-12-07-aspnetcore-authorization-support.markdown [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+- When setup, dynamic quota throttle acquisition.
+- Calls to registered `IDispatchMessageInspector.AfterReceiveRequest`
+- Concurrency lock acquisition
+
+Another impact is that authorization will now run on a captured `SynchronizationContext`. This point can impact CoreWCF services hosted in a UI thread (WPF or WinForms app).
+
+### Exclusiveness of ASP.NET Core Authorization policies and `ServiceAuthorizationManager`
+
+Having `ClientCredentialType` set to `InheritedFromHost` disable the execution of an authorization logic implemented in `ServiceAuthorizationManager`.
```
Comment (mconnew) [inline _posts/2022-12-07-aspnetcore-authorization-support.markdown]: disable -> disables

File: _posts/2022-12-07-aspnetcore-authorization-support.markdown [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+### `ServiceAuthenticationManager` and `ServiceAuthorizationManager` API modernization
+
+Both implementations now support asynchronous implementations. Existing synchronous implementations will still be compatible but have been deprecated and will trigger a build warning.
```
Comment (mconnew) [inline _posts/2022-12-07-aspnetcore-authorization-support.markdown]: These two classes now have async versions of the virtual methods which you can override. The existing synchronous method have been deprecated using the `Obsolete` attribute and will cause a build warning if you override them. If you are overriding one of the existing synchronous virtual methods, your code will continue to function the same as it always has and will continue to do so for all future 1.x releases. The synchronous variations of the methods will likely be removed in a future 2.x release. You can safely suppress the build warning until you have migrated your implementation to the async methods.

**Comment by g7ed6e** [comment]
TODO Add a link to the sample from https://github.com/CoreWCF/samples/pull/29 once merged

File: _posts/2022-12-07-aspnetcore-authorization-support.markdown (hunk 1 above)
Comment (g7ed6e) [reply _posts/2022-12-07-aspnetcore-authorization-support.markdown]: done

File: _posts/2022-12-07-aspnetcore-authorization-support.markdown (hunk 2 above)
Comment (g7ed6e) [reply _posts/2022-12-07-aspnetcore-authorization-support.markdown]: done

File: _posts/2022-12-07-aspnetcore-authorization-support.markdown (hunk 4 above)
Comment (g7ed6e) [reply _posts/2022-12-07-aspnetcore-authorization-support.markdown]: done

File: _posts/2022-12-07-aspnetcore-authorization-support.markdown (hunk 6 above)
Comment (g7ed6e) [reply _posts/2022-12-07-aspnetcore-authorization-support.markdown]: done

**Review by mconnew — APPROVED** [review]

**Pull request merged**